src/Elastic.Apm.SqlClient/SqlClientDiagnosticSubscriber.cs

```diff
@@ -9,13 +10,20 @@ public class SqlClientDiagnosticSubscriber : IDiagnosticsSubscriber
 		public IDisposable Subscribe(IApmAgent agentComponents)
 		{
 			var retVal = new CompositeDisposable();
-			var initializer = new DiagnosticInitializer(agentComponents.Logger, new[] { new SqlClientDiagnosticListener(agentComponents) });
+			if (PlatformDetection.IsDotNetCore)
```

Comment:
Are we totally sure about this? I tried to research and understand this and this seems to be correct, but is it always the case that on Full Framework all `SqlClient` versions emit through EventListener and on .NET Core through DiagnosticSource? 

Do you know anything about this?

Reply:
So, what I see in [`ApplicationInsights` repo](https://github.com/microsoft/ApplicationInsights-dotnet/blob/58618d7aaad5407d8fb5f6837fdc8e1096f88394/WEB/Src/DependencyCollector/Shared/DependencyTrackingTelemetryModule.cs#L27-L36). `EventListener` is used only in `net45` target. `DiagnosticSource` is used in both - `net45` and `netstandard` as well as for `HttpClient`.

As I understand correctly [the comment](https://github.com/microsoft/ApplicationInsights-dotnet/blob/58618d7aaad5407d8fb5f6837fdc8e1096f88394/WEB/Src/DependencyCollector/Shared/DependencyTrackingTelemetryModule.cs#L138)

Reply:
I see, thanks for the info.

Well, to me that comment in the AI code is a but cryptic :) I know that `HttpClient` was updated outside the framework with a [NuGet pacakge](https://www.nuget.org/packages/System.Net.Http/) - maybe that's what they mean by "referencing .NET Core System.Net.Http" - I suspect that package was built from the prior CoreFX repo - same for .NET Core. 

There isn't really such thing for `SqlClient` (unless you manually copy assemblies) - or actually  `Microsoft.Data.SqlClient` would be similar, but that seems to use EventListener on Full Framework - which surprised me - I though it'd use DiagnosticSource on both.

Is that maybe why Application Insights subscribe to DiagnosticSource on both?  I was unable to get DiagnosticSource out of this on Full Framework.

Nevertheless this is I'd say a minor thing. I think the current code is fine - if someone runs into a situation where we don't turn on the right stuff we can easily find that this is the root cause and we can adapt this part of the code.

What do you think? Keep it as it is?

Reply:
> I know that HttpClient was updated outside the framework with a NuGet pacakge - maybe that's what they mean by "referencing .NET Core System.Net.Http"

I also thought so, however, when I tried to do it, there was loaded usual NetFx version of `HttpClient`.

> I though it'd use DiagnosticSource on both
 
As I understand correctly, `DiagnoticSource` is built-in in .Net Core runtime, and as for .Net Framework, it can be added only from the NuGet package. It seems it's one of the main reasons, why `DiagnosticSource` isn't widely used in .Net Framework.

> What do you think? Keep it as it is?

I suggest keeping it as is, at least for now.